opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java

```diff
@@ -186,9 +201,165 @@ private JSONObject createFeatureCollection() {
+                        task.setCode(IRS);
+                        task.setDescription(applicationContext.getString(R.string.irs_task_description));
+                        task.setFocus(IRS_VISIT);
+                    } else if (Constants.StructureType.MOSQUITO_COLLECTION_POINT.equals(structureType)) {
+                        task.setCode(MOSQUITO_COLLECTION);
+                        task.setDescription(applicationContext.getString(R.string.mosquito_collection_task_description));
+                        task.setFocus(MOSQUITO_COLLECTION);
+                    } else if (Constants.StructureType.LARVAL_BREEDING_SITE.equals(structureType)) {
+                        task.setCode(LARVAL_DIPPING);
+                        task.setDescription(applicationContext.getString(R.string.larval_dipping_task_description));
+                        task.setFocus(LARVAL_DIPPING);
+                    }
+                    task.setForEntity(structure.getId());
+                    task.setExecutionStartDate(now);
+                    task.setAuthoredOn(now);
+                    task.setLastModified(now);
+                    task.setOwner(event.getProviderId());
+                    task.setSyncStatus(BaseRepository.TYPE_Created);
+                    taskRepository.addOrUpdate(task);
+                    appExecutors.mainThread().execute(new Runnable() {
+                        @Override
+                        public void run() {
+                            Map<String, String> taskProperties = new HashMap<>();
+                            taskProperties.put(Constants.Properties.TASK_IDENTIFIER, task.getIdentifier());
+                            taskProperties.put(Constants.Properties.TASK_BUSINESS_STATUS, task.getBusinessStatus());
+                            taskProperties.put(Constants.Properties.TASK_STATUS, task.getStatus().name());
+                            taskProperties.put(Constants.Properties.TASK_CODE, task.getCode());
+                            taskProperties.put(Constants.Properties.LOCATION_UUID, structure.getProperties().getUid());
+                            taskProperties.put(Constants.Properties.LOCATION_VERSION, structure.getProperties().getVersion() + "");
+                            taskProperties.put(Constants.Properties.LOCATION_TYPE, structure.getProperties().getType());
+                            structure.getProperties().setCustomProperties(taskProperties);
+                            presenterCallBack.onStructureAdded(Feature.fromJson(gson.toJson(structure)), featureCoordinates);
+                        }
+                    });
+                } catch (JSONException e) {
+                    Log.e(TAG, "Error saving new Structure", e);
+                    presenterCallBack.onFormSaveFailure(REGISTER_STRUCTURE_EVENT);
+                }
+            }
+        };
+
+        appExecutors.diskIO().execute(runnable);
+    }
+
+
+    private void saveMosquitoInterventionForm(JSONObject jsonForm) {
```

Comment:
Remove this method, method already defined in the parent class  `BaseInteractor`.
There are already methods saveMosquitoCollectionForm and `saveSprayForm` on `BaseIntercator`
Use those methods or refactor/merge them. All save methods should be on the `BaseIntercator` so that the functionality can be reused in the register views

Reply:
Done.